chemist-aws/src/main/scala/AwsDiscovery.scala

```diff
@@ -314,16 +325,19 @@ class AwsDiscovery(
 
     val datacenter = in.getPlacement.getAvailabilityZone
 
-    val mirrorTemplate: Option[String] =
-      machineTags.get(AwsTagKeys.mirrorTemplate) orElse defaultInstanceTemplate
+    //Note: if node is not labeled explicitly we ignore it to avoid wasting time trying to continiously
+    // reconnect to incompatible instances
+    val mirrorTemplate: Option[String] = machineTags.get(AwsTagKeys.mirrorTemplate)
 
-    for {
+    val r = for {
       a <- toLocation(dns, datacenter, mirrorTemplate, Mirroring)
-      // _  = log.debug(s"discovered mirrioring template '$a'")
+      // _  = log.debug(s"discovered mirroring template '$a'")
     } yield AwsInstance(
       id = in.getInstanceId,
       tags = machineTags,
       locations = NonEmptyList(a)
     )
+
+    r.leftMap(message => (in.getInstanceId, message))
```

Comment:
"What is the purpose of fromAWSInstance then" I'm not sure how the refactoring of what is happening inside the method affects this, but...

I would say something like "what is the point of a single line for comprehension" especially when the thing inside the for comprehension is a special purposed thing just to service this for comprehension, but anyway, this is not worth bickering about.